x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -177,56 +170,75 @@ func createCloudWatchEvents(getMetricDataResults []cloudwatch.MetricDataResult,
+				}
 
-	if len(mapOfMetricSetFieldResults) <= 11 {
-		info = "Missing Cloudwatch data for instance " + instanceID + ". This is expected for non-running instances or " +
-			"a new instance during the first data collection. If this shows up multiple times, please recheck the period " +
-			"setting in config."
-	}
+				monitoringState, err := instanceOutput[instanceID].Monitoring.State.MarshalValue()
+				if err != nil {
+					return events, "", errors.Wrap(err, "instance.Monitoring.State.MarshalValue failed")
+				}
 
-	instanceStateName, err := instanceOutput.State.Name.MarshalValue()
-	if err != nil {
-		err = errors.Wrap(err, "instance.State.Name.MarshalValue failed")
-		return
-	}
+				events[instanceID].MetricSetFields.Put("instance.image.id", *instanceOutput[instanceID].ImageId)
+				events[instanceID].MetricSetFields.Put("instance.state.name", instanceStateName)
+				events[instanceID].MetricSetFields.Put("instance.state.code", *instanceOutput[instanceID].State.Code)
+				events[instanceID].MetricSetFields.Put("instance.monitoring.state", monitoringState)
+				events[instanceID].MetricSetFields.Put("instance.core.count", *instanceOutput[instanceID].CpuOptions.CoreCount)
+				events[instanceID].MetricSetFields.Put("instance.threads_per_core", *instanceOutput[instanceID].CpuOptions.ThreadsPerCore)
+				publicIP := instanceOutput[instanceID].PublicIpAddress
+				if publicIP != nil {
+					events[instanceID].MetricSetFields.Put("instance.public.ip", *publicIP)
+				}
 
-	monitoringState, err := instanceOutput.Monitoring.State.MarshalValue()
-	if err != nil {
-		err = errors.Wrap(err, "instance.Monitoring.State.MarshalValue failed")
-		return
+				events[instanceID].MetricSetFields.Put("instance.public.dns_name", *instanceOutput[instanceID].PublicDnsName)
+				events[instanceID].MetricSetFields.Put("instance.private.dns_name", *instanceOutput[instanceID].PrivateDnsName)
+				privateIP := instanceOutput[instanceID].PrivateIpAddress
+				if privateIP != nil {
+					events[instanceID].MetricSetFields.Put("instance.private.ip", *privateIP)
+				}
+
+			}
+
+		}
 	}
 
-	resultMetricSetFields.Put("instance.image.id", *instanceOutput.ImageId)
-	resultMetricSetFields.Put("instance.state.name", instanceStateName)
-	resultMetricSetFields.Put("instance.state.code", *instanceOutput.State.Code)
-	resultMetricSetFields.Put("instance.monitoring.state", monitoringState)
-	resultMetricSetFields.Put("instance.core.count", *instanceOutput.CpuOptions.CoreCount)
-	resultMetricSetFields.Put("instance.threads_per_core", *instanceOutput.CpuOptions.ThreadsPerCore)
-	publicIP := instanceOutput.PublicIpAddress
-	if publicIP != nil {
-		resultMetricSetFields.Put("instance.public.ip", *publicIP)
+	infoPrefix := "Missing Cloudwatch data, this is expected for non-running instances" +
+		" or a new instance during the first data collection. If this shows up" +
+		" multiple times, please recheck the period setting in config. Instance ID: "
+	infoID := ""
```

Comment:
What do you think about logging this directly in this function instead of returning the message? you can make this function a method of the metricset to have access to `Logger()`

Reply:
yes, didn't know why it was not done previously 😬  Thanks for the suggestion. Now the code looks a lot cleaner here....